tests/AutoReview/ProjectCodeTest.php

```diff
@@ -500,6 +502,85 @@ static function ($className) {
         );
     }
 
+    /**
+     * @dataProvider provideAllCodeIsMonolithicPhpCases
+     *
+     * @param string $file
+     */
+    public function testAllCodeIsMonolithicPhp($file)
+    {
+        $tokens = Tokens::fromCode(file_get_contents($file));
+
+        static::assertTrue($tokens->isMonolithicPhp(), sprintf('File "%s" is not monolithic.', $file));
+        static::assertTrue($tokens->isAnyTokenKindsFound([T_CLASS, T_TRAIT, T_INTERFACE]), sprintf('File "%s" does not contain a classy.', $file));
+
+        $classes = $tokens->findGivenKind(T_CLASS);
+        $interfaces = $tokens->findGivenKind(T_INTERFACE);
+        $traits = $tokens->findGivenKind(T_TRAIT);
+
+        if (\count($interfaces) > 0) {
+            static::assertCount(1, $interfaces, 'There should only be one interface definition in a file.');
+            static::assertCount(0, $classes, 'A file with interface definition should not contain classes.');
+            static::assertCount(0, $traits, 'A file with interface definition should not contain traits.');
+
+            return;
+        }
+
+        $numberOfTraitDefinitions = 0;
+
+        if (\count($traits) > 0) {
+            foreach ($traits as $index => $trait) {
+                $previous = $tokens->getPrevMeaningfulToken($index);
+
+                if (null !== $previous && !$tokens[$previous]->isGivenKind(CT::T_USE_TRAIT)) {
+                    ++$numberOfTraitDefinitions;
+                }
+            }
+        }
+
+        $numberOfClassDefinitions = 0;
+
+        if (\count($classes) > 0) {
+            $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+            foreach ($classes as $index => $class) {
+                if ($tokensAnalyzer->isAnonymousClass($index)) {
+                    continue;
+                }
+
+                ++$numberOfClassDefinitions;
+            }
+        }
+
+        if ($numberOfTraitDefinitions > 0) {
+            static::assertSame(1, $numberOfTraitDefinitions, sprintf('File "%s" contains multiple trait definitions.', $file));
+            static::assertSame(0, $numberOfClassDefinitions, sprintf('File "%s" contains a trait definition and a class definition.', $file));
+        }
+
+        if ($numberOfClassDefinitions > 0) {
+            if (false === strpos($file, 'PHP-CS-Fixer/src/Event/Event.php') && false === strpos($file, 'PHP-CS-Fixer/tests/TestCase.php')) {
+                static::assertSame(1, $numberOfClassDefinitions, sprintf('File "%s" contains multiple class definitions.', $file));
+            }
+
+            static::assertSame(0, $numberOfTraitDefinitions, sprintf('File "%s" contains a class definition and a trait definition.', $file));
+        }
+
+        static::assertGreaterThan(0, $numberOfTraitDefinitions + $numberOfClassDefinitions, sprintf('File "%s" does not contains a class, interface or trait definition.', $file));
+    }
+
+    public function provideAllCodeIsMonolithicPhpCases()
```

Comment:
Do we need this provider? Can't we use `provideSrcClassCases` and `provideTestClassCases`?